src/bittybuzz/include/bittybuzz/BittyBuzzLib.h

```diff
@@ -0,0 +1,33 @@
+#ifndef __BITTYBUZZSTDLIB_H_
+#define __BITTYBUZZSTDLIB_H_
+
+#include "IBittyBuzzLib.h"
+#include <cstdint>
+#include <optional>
+
+template <typename Container>
```

Comment:
There is an interface, IBittyBuzzlLib. the template is only for the container of BittyBuzzLibMemberRegister so we can store it somewhere. I could have done a static array and then copy it, though this array would have a max size. By using template, the object will be exactly the size it needs to be, not more, not less

Reply:
From what I understand, you do a for each on the elements of the container. It's just a bit confusing since as you said, I could be trying to set a static array in the template, but since you are using a for each in your code it would not work. Having an interface would make it more explicit to the user that the Container is expected to have .begin() and .end() of type BittyBuzzLibMemberRegister.

It's more of a nitpick, the template makes the usage a bit complicated and lacks transparency.